opslevel/datasource_opslevel_rubric_categories.go

```diff
@@ -1,50 +1,88 @@
 package opslevel
 
-// import (
-// 	"github.com/opslevel/opslevel-go/v2024"
-
-// 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
-// )
-
-// func datasourceRubricCategories() *schema.Resource {
-// 	return &schema.Resource{
-// 		Read: wrap(datasourceRubricCategoriesRead),
-// 		Schema: map[string]*schema.Schema{
-// 			"ids": {
-// 				Type:     schema.TypeList,
-// 				Computed: true,
-// 				Elem:     &schema.Schema{Type: schema.TypeString},
-// 			},
-// 			"names": {
-// 				Type:     schema.TypeList,
-// 				Computed: true,
-// 				Elem:     &schema.Schema{Type: schema.TypeString},
-// 			},
-// 		},
-// 	}
-// }
-
-// func datasourceRubricCategoriesRead(d *schema.ResourceData, client *opslevel.Client) error {
-// 	result, err := client.ListCategories(nil)
-// 	if err != nil {
-// 		return err
-// 	}
-
-// 	count := len(result.Nodes)
-// 	aliases := make([]string, count)
-// 	ids := make([]string, count)
-// 	indexes := make([]int, count)
-// 	names := make([]string, count)
-// 	for i, item := range result.Nodes {
-// 		ids[i] = string(item.Id)
-// 		names[i] = item.Name
-// 	}
-
-// 	d.SetId(timeID())
-// 	d.Set("aliases", aliases)
-// 	d.Set("ids", ids)
-// 	d.Set("indexes", indexes)
-// 	d.Set("names", names)
-
-// 	return nil
-// }
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+// Ensure CategoryDataSourcesAll implements DataSourceWithConfigure interface
+var _ datasource.DataSourceWithConfigure = &CategoryDataSourcesAll{}
+
+func NewCategoryDataSourcesAll() datasource.DataSource {
+	return &CategoryDataSourcesAll{}
+}
+
+// CategoryDataSourcesAll manages a Category data source.
+type CategoryDataSourcesAll struct {
+	CommonDataSourceClient
+}
+
+// categoryDataSourceModel describes the data source data model.
+type categoryDataSourceModel struct {
+	Id   types.String `tfsdk:"id"`
+	Name types.String `tfsdk:"name"`
+}
+
+// CategoryDataSourcesModel describes the data source data model.
+type CategoryDataSourcesModel struct {
+	RubricCategories []categoryDataSourceModel `tfsdk:"rubric_categories"`
+}
+
+func NewCategoryDataSourcesModel(categories []opslevel.Category) CategoryDataSourcesModel {
+	rubricCategories := []categoryDataSourceModel{}
+	for _, category := range categories {
+		rubricCategory := categoryDataSourceModel{
```

Comment:
This is fine for me, but I wonder if in the future what we could do is just use the singular data source and just call `NewCategoryDataSourceModel()` and append it to the list instead of having to repeat the instantiation logic here.

Reply:
@davidbloss maybe for the plural datasources, we could do `NewMultiCategoryDataSourceModel()`